gossipd/routing.h

```diff
@@ -198,9 +199,8 @@ struct routing_state {
 	/* node_announcements which are waiting on pending_cannouncement */
 	struct pending_node_map *pending_node_map;
 
-	/* FIXME: Make this a htable! */
 	/* channel_announcement which are pending short_channel_id lookup */
-	struct list_head pending_cannouncement;
+	struct pending_cannouncement_map *pending_cannouncements;
```

Comment:
Nice!  Only one trivial change: we don't need a pointer here, saving an allocation.

Reply:
Thank you! I've done it :)

BTW, when shouldn't I use the pointer for a htable?
There `nodes` and `pending_node_map` are both the pointer to htable. Should I also save them as allocations?
```
struct routing_state {
...
	struct node_map *nodes;

	struct pending_node_map *pending_node_map;

	struct pending_cannouncement_map pending_cannouncements;
...
};
```

Reply:
Yes!  Unless you try to convert it and realize there's some subtle reason I'm wrong :)